portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,7 +20,7 @@
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
         String id = request.getParameter(PropertyNames.VIDEO_ID);
-        if(id == null){
+        if(id == null || id == ""){
```

Comment:
Strings are objects in Java, so we should perform a comparison like .equals("").
Would recommend using String.isEmpty().